### PR TITLE
[2.13.x] DDF-4367 Fix search results merging in Intrigue

### PR DESCRIPTION
**Pull request by mdang8**
#### What does this PR do?

Back-ports #4092 

Fixes an issue when making searches in Intrigue where there are 250+ results returned. The counts at the bottom of the result list now match with the expected number of results. The queued results are changed to reset instead of being added because the add operation is slow enough that it can cause partial merges to happen and results being removed. The issue was more commonly observed on Firefox browsers than Chrome because the operation was finishing outside of the default auto-merge time threshold.

#### Who is reviewing it? 

@emmberk @kcover @peterhuffer @pvargas @brianfelix 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@clockard

#### How should this be tested?

- Open Intrigue in a Chrome browser
- Ingest 265 (or any number greater than 250) metacards using the UI uploader
- Start a wildcard search and verify that the results text at the top query details box reads `265 results`
- Verify that the results label text at the bottom of the results list reads `250 results` or `1-25 of 265`
- Click the arrow button to proceed to the next pages until it reaches the end and the button changes to `More`
- Verify that the results label text reads `226-250 of 265`
- Click the `More` button and verify that the results label text at the bottom of the results list reads `15 results` or `251-265 of 265`
- Re-run the search and verify that the results are consistent with the above
- Remove the uploaded metacards through the console
- Upload 600 (or any number greater than 500) new metacards, repeat the process, and verify it still works
- Duplicate the same process in a Firefox browser and verify that everything works

#### What are the relevant tickets?
[DDF-4367](https://codice.atlassian.net/browse/4367)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by kcover** [comment]
✏️ Your commit message could be more informative. We have some guidelines here you could check out https://codice.atlassian.net/wiki/spaces/DDF/pages/6291518/Git+Commit+Message+Guidelines. It's also good to squash your commits into one when the PR is relatively small. That keeps the commit log concise so it doesn't have a bunch of commits that say "addressing comments", "reformatted", "fixed header" etc.

**Comment by pvargas** [comment]
Hero failed.
I followed the instructions and got the results below.
screen shot 2018-12-21 at 3 07 08 pm 3

**Comment by clockard** [comment]
build noq

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js [hunk 1]
```diff
@@ -212,23 +212,37 @@ module.exports = Backbone.AssociatedModel.extend({
...
+    },
     allowAutoMerge: function () {
         if (this.get('results').length === 0 || !this.get('currentlyViewed')) {
             return true;
         } else {
             return (Date.now() - this.lastMerge) < properties.getAutoMergeTime();
         }
     },
-    mergeQueue: function (userTriggered) {
+    mergeQueue: function (userTriggered, includeQueuedCache = true) {
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js]: I think we need to look into why we need the logic for the queued cache.  I'm hoping we can remove it from master.

Comment (mdang8) [reply ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js]: I can remove the part with the queued cache results and come up with an alternative solution for that.

**Comment by vinamartin** [comment]
❓ Is this port still actively being worked?

**Pull request closed without merging**